### PR TITLE
Rebrand the Xenia window with module metadata

### PR DESCRIPTION
**Pull request by x1nixmzeng**
Replaces window title and icon with the resources in the current module - part of #506 

Example:

limbo_example

### CODE REVIEW

**Comment by x1nixmzeng** [comment]
Also (as I don't have a voice on #xenia-dev for whatever reason :disappointed:):

* Do you have any suggestions on where to store off the metadata for other uses (save system, etc)
* Is it too early to allow setting the region? (and provide regional game names)

**Comment by DrChat** [comment]
Ooh, this is really cool! Also, to talk on #xenia-dev, you just need to be authenticated by NickServ.

**Comment by rlabrecque** [comment]
I don't like that the title no longer says xenia and doesn't contain the branch info anymore, otherwise I love it.

**Comment by Razzile** [comment]
@rlabrecque maybe it would be a good idea for rolling builds to show build number but actual releases to revert to game metadata

**Comment by x1nixmzeng** [comment]
I considered creating a flag which allows the title to be completely replaced.

For now, there's a silly bug where `s` isn't incremented if the XDBF isn't found. I also want to figure out some other headers, as there should be achivements somewhere.

File: src/xenia/ui/window.h [hunk 1]
```diff
@@ -53,6 +53,8 @@ class Window {
     return true;
   }
 
+  virtual bool set_icon_from_buffer(void *buffer, size_t size) = 0;
```
Comment (DrChat) [inline src/xenia/ui/window.h]: This should be named SetIconFromBuffer. Only use lowercase underscored names for functions that directly return a variable.

**Comment by x1nixmzeng** [comment]
Latest:

xenia-2

Off topic, I don't understand the linting errors. I'm staging and running `.\xb.bat format` before committing changes, but then travis rearranges the pointers? Do we not share the same formatting rules?

**Comment by DrChat** [comment]
About that - @benvanik, why not automatically lint pull requests on merge?

Also @x1nixmzeng, there's an automatic linter for VS that I use. It's documented in the style guide.

**Comment by DrChat** [comment]
:+1:

**Comment by benvanik** [comment]
Cool stuff!

Will land and clean up some style stuff.

The linter on travis is a weird version - no longer matches head, and I haven't wanted to have our build process build clang from source :/ hoping to stabilize on 3.8 (which in theory is released in feb)

**Pull request merged**